### ISSUE TITLE
Investigate if it is possible for JSON to avoid base64 in js/json

### ISSUE THREAD

**Issue by Firstyear** [kanidm/webauthn-rs#5]
We currently deviate from the standard by base64 encoding some JSON members in challenges and responses. This may confuse some consumers, but this difficulty comes in from ArrayBuffers being a `'a [u8]` in a struct, which causes lifetime issues with actix. I have also noted odd behaviours where the output of navigator.credentials methods can't be sent to clients either, also causing us to do some odd js tricks.

Basically, we need to improve the js parts, and work out better ways to handle this aspect of the system.

**Comment by aseigler** [comment]
For conformance particularly, assume everyone wants base64url encoding without padding.

**Comment by Firstyear** [comment]
The spec was unclear (or I missed it?) about the base64 as url-safe or not. At the moment I attempt both url and standard, and fail only if both fail. I've opened a seperate issue to handle this issue.

This issue is more about the fact that there are a number of ArrayBuffer types that I just can't seem to handle nicely with Serde, so as a result my client js and server are base64'ing data that perhaps should not be. For example, the rawId in authenticate I am base64 encoding, making it equivalent to Id field.

So to be properly compliant with "any javascript client" parts, I should not require base64 on these arrayBuffer types incorrectly. :)

**Comment by Firstyear** [comment]
Closing because it's not possible to create json types that can become a Uint8Array on a browser. You *must* base64 these datafields. It surprises me that the webauthn standard does not just say "base64 these" and then leave it to the navigator.credentials api to handle it, but alas, instead all javascript implementations now have the extreme joy of working out how to get from base64 to a uint8Array and tampering with the internals of any server issued request.
